pandas/core/arrays/datetimes.py

```diff
@@ -681,6 +685,43 @@ def _format_native_types(
             self.asi8, tz=self.tz, format=fmt, na_rep=na_rep
         )
 
+    def _format_array(
+        self,
+        formatter: Callable | None,
```

Comment:
im not too familiar with how this gets reached.  is e.g. formatter going to always be self._formatter?

Reply:
I think you're right.

Reply:
then can we do without the formatter arg?

Reply:
Sorry I was incorrect. `formatter` might be a user-supplied callable in cases like `df.to_html(formatter={"col": formatter})`.